src/touch/gestures.c

```diff
@@ -94,6 +102,8 @@ static void _slider_state_update(gestures_detection_state_t* state, uint16_t pos
     state->velocity_sum = state->velocity_sum - velocity_removed + velocity_current;
     state->velocity_values[state->velocity_index] = velocity_current;
     state->velocity_index = (state->velocity_index + 1) % MAX_HISTORY;
+    uint16_t distance_from_start = abs((int)position - (int)state->position_start);
```

Comment:
Is the `int` typecast needed? (or if not needed, then considered safer) 
If so, check for other places using `abs()` which should then also add the typecast (e.g. a couple lines below).

Reply:
My typecast expertise is a bit rusty - wouldn't the case in which state->position > position underflow the uint to <big number>, which then gets casted to int? As in the evaluation order, the subtraction will come first.

Reply:
C implicit casting is very non-intuitive. It might just be that the operands are implicitly cast as the resulting expression is cast to `int`, but I'm also not sure anymore.

I prefer explicit casting no matter the rules, for clarity.

Reply:
Yep I'll leave explicit casting to the places in which it's unsure what the signedness of a subtraction might be.